base/abstractarraymath.jl

```diff
@@ -415,7 +415,7 @@ _reperr(s, n, N) = throw(ArgumentError("number of " * s * " repetitions " *
                 n = inner[i]
                 inner_indices[i] = (1:n) + ((c[i] - 1) * n)
             end
-            R[inner_indices...] = A[c]
+            fill!(view(R, inner_indices...), A[c])
```

Comment:
This must be the cause of the repeat regression, @pabloferz

Reply:
This is a necessary fix, it's meant for the case when `A[c]` is itself an array. I can think of a way of trying to preserve the speed for the cases when `A[c]` is not an array, but I don't know what would be the action to take with regards to the backport.

Reply:
https://github.com/JuliaLang/julia/pull/24022